src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/JsonDictionaryConverter.cs

```diff
@@ -114,10 +116,12 @@ protected static JsonConverter<T> GetConverter<T>(JsonTypeInfo typeInfo)
                         // Read method would have thrown if otherwise.
                         Debug.Assert(reader.TokenType == JsonTokenType.PropertyName);
 
-                        TKey key = ReadDictionaryKey(ref reader, ref state);
+                        state.Current.JsonPropertyInfo = keyTypeInfo.PropertyInfoForTypeInfo;
```

Comment:
Ensures that the right `JsonPropertyInfo` is surfaced depending on whether we serialize the key or the value for the dictionary.

Reply:
Is this covered by tests? I didn't see any "property name + exception" test changes here that were dictionary-specific.

Reply:
Yes. See the changes to the `ThrowOnInvalidFormat` test in the first commit: https://github.com/dotnet/runtime/pull/67109/commits/fb35427bfeb0aaae03896c56f833c43c705e2c8d#diff-e978d8822ab16e2ef9eb42029b8cfe1bb27717da7f7b6702ac864de3fd5c3a38